source/fab/grabber.py

```diff
@@ -0,0 +1,65 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+"""
+Core of the source extraction tool.
+"""
+import logging
+from pathlib import Path
+from typing import Sequence
+
+from fab.repository import Repository, repository_from_url
+
+
+def entry() -> None:
+    """
+    Extract and merge up a source tree from several repositories.
+    """
+    import argparse
+    import sys
+    import fab
+
+    logger = logging.getLogger('fab-grab')
```

Comment:
Related to my earlier comment about `fab-dumper` - the logging here uses the name of the command-line tool rather than being `fab-grabber` (personally I think I prefer this way; so it corresponds to the name of the typed command)